caput/scripts/runner.py

```diff
@@ -123,43 +124,18 @@ def run(configfile, loglevel, profile, profiler):
             "--loglevel is deprecated, use the config file instead", DeprecationWarning
         )
 
-    if profile:
-        if profiler == "cProfile":
-            import cProfile
-
-            pr = cProfile.Profile()
-            pr.enable()
-        elif profiler == "pyinstrument":
-            from pyinstrument import Profiler
-
-            pr = Profiler()
-            pr.start()
-
-    try:
-        P = Manager.from_yaml_file(configfile)
-    except CaputConfigError as e:
-        click.echo(
-            "Found at least one error in '{}'.\n"
-            "Fix and run again to find more problems.".format(configfile)
-        )
-        click.echo(e)
-        sys.exit(1)
-    else:
-        P.run()
-
-    if profile:
-
-        from caput import mpiutil
-
-        rank = mpiutil.rank
-
-        if profiler == "cProfile":
-            pr.disable()
-            pr.dump_stats("profile_%i.prof" % mpiutil.rank)
-        elif profiler == "pyinstrument":
-            pr.stop()
-            with open("profile_%i.txt" % rank, "w") as fh:
-                fh.write(pr.output_text(unicode=True))
+    with Profiler(profile, profiler=profiler):
```

Comment:
How would `IOUsage` be used here? Maybe I missed sth

Reply:
Yeah, it doesn't get used there. I just have a few blocks in driftscan that I want to see how much IO is done and so I wrap them in a block like:
```python
with IOUsage(logger=logger):
    bt.generate_mfiles()
```